packages/client/src/views/RecordAudit/RecordAudit.tsx

```diff
@@ -598,6 +597,14 @@ const BodyContent = ({
       status: wqStatus || draftStatus
     }
 
+    if (
+      draft?.assignmentStatus?.createdAt &&
+      workqueueDeclaration?.registration?.assignment?.createdAt &&
+      workqueueDeclaration?.registration?.assignment?.createdAt >
+        draft?.assignmentStatus?.createdAt
+    ) {
+      declaration.assignment = workqueueDeclaration.registration.assignment
+    }
```

Comment:
Could you add a comment here explaining the scenario and why this fixes it please?